CleverPush/Source/CleverPushInstance.m

```diff
@@ -1710,7 +1712,31 @@ - (void)didRegisterForRemoteNotifications:(UIApplication*)app deviceToken:(NSDat
 }
 
 #pragma mark - Generalised Api call.
+- (void)enqueueFailedRequest:(NSURLRequest *)request withRetryCount:(NSInteger)retryCount onSuccess:(CPResultSuccessBlock)successBlock onFailure:(CPFailureBlock)failureBlock {
+    NSURLSession *session = [NSURLSession sharedSession];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        if (successBlock != nil && error == nil) {
+            [self handleJSONNSURLResponse:response data:data error:error onSuccess:successBlock onFailure:failureBlock];
+        } else {
+            if (retryCount < 3) {
+                dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1.0 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
```

Comment:
This means we always wait 1 sec between the retries?

Instead please use exponential backoff (always multiply by retryCount by 2 = wait seconds)